package/create_source_tarball.sh

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
+#
+# Creates source tarballs for SuperCollider in the form of
+# 'SuperCollider-x.x.x-Source.tar.bz2' and moves the file to the location of
+# this script (the repository's package folder).
+#
+# Requirements:
+# - git
+# - a writable (user) /tmp folder for mktemp
+# - gnupg >= 2.0.0 (if source tarball signing is requested)
+# - a valid PGP key in the keyring (if source tarball signing is requested)
+
+set -euo pipefail
+
+get_absolute_path() {
+  cd "$(dirname "$1")" && pwd -P
+}
+
+validate_project_tag() {
+  if ! git ls-remote -t "${upstream}"| grep -e "${version}$" > /dev/null; then
+    echo "The tag '$version' could not be found in upstream repository (${upstream})."
+    exit 1
+  fi
+}
+
+checkout_project() {
+  echo "Cloning project below working directory ${working_dir}"
+  cd "$working_dir"
+  git clone "$upstream" --branch "$version" \
+                        --single-branch \
+                        --recursive \
+                        "${output_name}"
+}
+
+clean_sources() {
+  cd "${working_dir}/${output_name}"
+  echo "Removing unneeded files and folders..."
+  rm -rfv .clang* \
+          .editor* \
+          .git* \
+          .travis* \
+          .github* \
+          .appveyor* \
```

Comment:
would recommend simplifying this to `.*`, it took me a bit to realize it is equivalent

Reply:
True. I thought being a bit more verbose would be good, but in this case it serves no purpose.

Reply:
I just realized, that this makes rm exit with non-zero exit code:

```
rm: refusing to remove '.' or '..' directory: skipping './.'
rm: refusing to remove '.' or '..' directory: skipping './..'
```

Guess I'll have to guard that to not make the entire script fail.

Reply:
Hm, guarding it (i.e. `set +e` and then `set -e`) just makes it not delete things.
I'll revert to what I did before.
Also `rm -rf .*` is not entirely safe on all shells.

Reply:
fair points, what you had originally works then!